landing_page/spec/controllers/micropost_controller_spec.rb

```diff
@@ -0,0 +1,97 @@
+require 'rails_helper'
+
+RSpec.describe MicropostsController, type: :controller do
+  before :each do
+    @user = FactoryBot.create(:user)
+    session[:user_id] = @user.id
+    login(@user)
+  end
+
+  before :each do
+    allow(controller).to receive(:current_user).and_return(@user)
+    @micropost = FactoryBot.create(:micropost, user_id: @user.id)
+  end
```

Comment:
Can you see any way this could potentially be simplified? Also, I'm not sure if it's the case here, but do we need all of these variable for every single test? Could there be reasons that we wouldn't want to necessary make multiple models for every test?

Reply:
Good questions. There is room to refactor the code so it's in one block.  I think having these variables for each test is necessary. In order to create a post, the user needs to be logged in as the user id is a required field and simply defining a user's id when creating a post didn't work